README.md

```diff
@@ -58,53 +58,61 @@ type Config struct {
 	// the dials tag can be used as an alias so when the name in the config file
 	// changes, the code doesn't have to change.
 	Val2 int `dials:"val_2"`
-	// Dials follows the Go convention for flags and will look for the dials
-	// tag or field name in lower-kebab-case. Without any struct tags, it
-	// would look for val-3 flag. To specify a different flag name, use the
-	// `dialsflag` tag. Now, Dials will lookup "some-val" flag instead.
-	// Since only `dialsflag` tag is specified, Val3 will only be populated
-	// from command line flags
-	Val3 bool `dialsflag:"some-val"`
+	// Dials will register a flag with the name matching the dials tag.
+	// Without any struct tags, dials will decode the go camel case field name
+	// and encode it using lower-kebab-case and use the encoded name as the flag
+	// name (ex: val-3). To specify a different flag name, use the `dialsflag`
+	// tag. Now, Dials will register a flag with "some-val" name instead.
+	// The `dialsdesc` tag is used to provide help message for the flag.
+	Val3 bool `dialsflag:"some-val" dialsdesc:"enable auth"`
 	// Path holds the value of the path to the config file. Dials follows the
-	// Go convention and will look for the dials tag or field name in all caps
-	// when struct tags aren't specified. Without any struct tags, it would
-	// lookup the PATH environment variable. To specify a different env variable,
-	// use the `dials_env` tag. Now Dials will lookup "configpath" env value to
-	// populate the Path field
+	// Go convention for environment variables and will look for the dials tag
```

Comment:
Let's make this the `*nix convention for environment variables`. (Screaming snake case is a *nix thing, not a Go thing.)

(with the exception of Java, but that's just wrong)